models/search.js

```diff
@@ -0,0 +1,104 @@
+import database from 'infra/database';
+import pagination from 'models/pagination.js';
+
+function generateTsQuery(query) {
+  const cleanedSearchTerm = decodeURIComponent(query.toLowerCase().trim());
+  const escapedSearchTerm = cleanedSearchTerm.replace(/'/g, "''");
+  return `websearch_to_tsquery('portuguese', '${escapedSearchTerm}')`;
+}
+
+function generateOrderByClause(strategy, tsQuery) {
+  switch (strategy) {
+    case 'new':
+      return 'ORDER BY published_at DESC';
+    case 'relevant':
+      return `ORDER BY CASE WHEN ts_rank(search, ${tsQuery}) > 0.6 THEN ts_rank(search, ${tsQuery}) ELSE 0 END DESC`;
+    case 'old':
+      return 'ORDER BY published_at ASC';
+    default:
+      return '';
+  }
+}
+
+async function findAll(values = {}) {
+  const offset = (values.page - 1) * (values.per_page || 10);
+  const limit = values.limit || values.per_page || 10;
+  const tsQuery = generateTsQuery(values.q);
+  const orderByClause = generateOrderByClause(values.strategy, tsQuery);
+  const withChildrenClause = values.only_children
+    ? 'AND contents.parent_id IS NOT NULL'
+    : values.with_children
+      ? ''
+      : 'AND contents.parent_id IS NULL';
+  const bodyClause = values.only_children || values.with_children ? " || ' ' || contents.body" : '';
+
+  const query = {
+    text: `
+      WITH content_window AS (
+        SELECT
+          COUNT(*) OVER()::INTEGER AS total_rows,
+          contents.id
+        FROM 
+          contents
+        INNER JOIN
+          users ON contents.owner_id = users.id
+        WHERE
+          to_tsvector('portuguese', COALESCE(contents.title, '') || ' ' || contents.slug || ' ' || users.username ${bodyClause}) @@ ${tsQuery}
+          AND contents.status = 'published'
+          ${withChildrenClause}
+          ${orderByClause}
+        LIMIT $1 OFFSET $2
+      )
+      SELECT
+        contents.id,
+        contents.owner_id,
+        contents.parent_id,
+        contents.slug,
+        contents.title,
+        ${values.only_children || values.with_children ? 'contents.body,' : ''}
```

Comment:
Apenar reforçando o outro comentário sobre o `body`:

```suggestion
        contents.body,
```